internal/armada/server/submit_to_log.go

```diff
@@ -361,6 +362,70 @@ func (srv *PulsarSubmitServer) CancelJobs(ctx context.Context, req *api.JobCance
 	}, nil
 }
 
+func (srv *PulsarSubmitServer) CancelJobSet(ctx context.Context, req *api.JobSetCancelRequest) (*api.CancellationResult, error) {
+	if req.Queue == "" {
+		return nil, &armadaerrors.ErrInvalidArgument{
+			Name:    "Queue",
+			Value:   req.Queue,
+			Message: "queue is empty",
+		}
+	}
+	if req.JobSetId == "" {
+		return nil, &armadaerrors.ErrInvalidArgument{
+			Name:    "JobSetId",
+			Value:   req.JobSetId,
+			Message: "JobSetId is empty",
+		}
+	}
+
+	err := validation.ValidateJobSetFilter(req.Filter)
+	if err != nil {
+		return nil, err
+	}
+
+	userId, groups, err := srv.Authorize(ctx, req.Queue, permissions.CancelAnyJobs, queue.PermissionVerbCancel)
+	if err != nil {
+		return nil, err
+	}
+
+	ids, err := srv.SubmitServer.jobRepository.GetJobSetJobIds(req.Queue, req.JobSetId, createJobSetFilter(req.Filter))
+	if err != nil {
+		return nil, status.Errorf(codes.Unavailable, "error getting job IDs: %s", err)
+	}
+
+	sequence := &armadaevents.EventSequence{
+		Queue:      req.Queue,
+		JobSetName: req.JobSetId,
+		UserId:     userId,
+		Groups:     groups,
+		Events:     make([]*armadaevents.EventSequence_Event, 0, len(ids)),
+	}
+
+	for _, id := range ids {
+		jobId, err := armadaevents.ProtoUuidFromUlidString(id)
+		if err != nil {
+			return nil, err
+		}
+
+		sequence.Events = append(sequence.Events, &armadaevents.EventSequence_Event{
+			Event: &armadaevents.EventSequence_Event_CancelJob{
+				CancelJob: &armadaevents.CancelJob{JobId: jobId},
+			},
+		})
+	}
+
+	err = srv.publishToPulsar(ctx, []*armadaevents.EventSequence{sequence})
+
+	if err != nil {
+		log.WithError(err).Error("failed send to Pulsar")
+		return nil, status.Error(codes.Internal, "Failed to send message")
```

Comment:
I think we could give a better error message to the user here- I.e "error sending cancel job messages to pulsar" or something.  It'll make our lives easier when we get reports of failure.

Reply:
Agreed, I just used what was already used for cancelling jobs